zookeeper-server/src/main/java/org/apache/zookeeper/server/SyncRequestProcessor.java

```diff
@@ -117,48 +118,46 @@ public void run() {
                 if (si == requestOfDeath) {
                     break;
                 }
-                if (si != null) {
```

Comment:
This looks like a lot of code change, but I simply removed this check 'is null' check and subsequently had to fix the indentation of the large block of code beneath it.